### PR TITLE
Pass directives to C backend in given order (#4078)

### PR DESCRIPTION
**Pull request by asandroq**
Directives for compiling and linking C code are order-sensitive and
therefore the order given in the input files must be preserved.

### CODE REVIEW

File: test/directives003/expected [hunk 1]
```diff
@@ -0,0 +1,6 @@
+nuyahcha
+mohgiide
+apeekair
+eeraimam
+rahhahwi
+goocheiv
```
Comment (ahmadsalim) [inline test/directives003/expected]: Are these just random words or do they mean something in some language?

**Comment by ahmadsalim** [comment]
Thanks for the PR. I will try to fix travis and hopefully your PR should succeed

File: test/directives003/expected (hunk 1 above)
Comment (asandroq) [reply test/directives003/expected]: They are random words generated with `pwgen -AB0`.

**Comment by ahmadsalim** [comment]
It passes now.
Can you please update the CHANGELOG and CONTRIBUTORS if needed?

I will merge afterwards

**Pull request merged**